fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java

```diff
@@ -474,6 +481,47 @@ private List<Column> genMaterializedViewColumns(CreateMaterializedViewStatement
             return reorderedColumns;
         }
 
+        private List<Index> genMaterializedViewIndexes(CreateMaterializedViewStatement statement) {
+            List<IndexDef> indexDefs = statement.getIndexDefs();
+            List<Index> indexes = statement.getMvIndexes();
+            List<Column> columns = statement.getMvColumnItems();
+
+            if (CollectionUtils.isNotEmpty(indexDefs)) {
+                Set<String> distinct = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+                Set<List<String>> distinctCol = new HashSet<>();
+
+                for (IndexDef indexDef : indexDefs) {
+                    indexDef.analyze();
+                    for (String indexColName : indexDef.getColumns()) {
+                        boolean found = false;
+                        for (Column column : columns) {
+                            if (column.getName().equalsIgnoreCase(indexColName)) {
+                                indexDef.checkColumn(column, statement.getKeysType());
+                                found = true;
+                                break;
+                            }
+                        }
+                        if (!found) {
+                            throw new SemanticException("BITMAP column does not exist in table. invalid column: " + indexColName,
+                                    indexDef.getPos());
+                        }
+                    }
+                    indexes.add(new Index(indexDef.getIndexName(), indexDef.getColumns(), indexDef.getIndexType(),
+                            indexDef.getComment()));
+                    distinct.add(indexDef.getIndexName());
+                    distinctCol.add(indexDef.getColumns().stream().map(String::toUpperCase).collect(Collectors.toList()));
+                }
+                if (distinct.size() != indexes.size()) {
+                    throw new SemanticException("index name must be unique", indexDefs.get(0).getPos());
+                }
+                if (distinctCol.size() != indexes.size()) {
+                    throw new SemanticException("same index columns have multiple index name is not allowed",
+                            indexDefs.get(0).getPos());
+                }
+            }
+            return indexes;
+        }
+
         private void checkExpInColumn(CreateMaterializedViewStatement statement) {
             ExpressionPartitionDesc expressionPartitionDesc = statement.getPartitionExpDesc();
             List<Column> columns = statement.getMvColumnItems();
```

Comment:
Avoid Mutable Parameters: The indexes list seems to be both created from the CreateMaterializedViewStatement's getter (getMvIndexes) and also modified (using add). That's a bit unclear and might lead to unintended side effects if it's used elsewhere in the project. Consider creating a local list within the function to collect new indexes, and then return that. Let the calling function decide how they want to handle merging of old and new indexes.

Reply:
Inconsistent Error Reporting: In error messages, you're referring to the position using indexDefs.get(0).getPos(). This could be misleading if the error is due to multiple indexDefs. You may want to set this in each loop iteration and use it in the exception.

Reply:
[fix gen bitmap index detail message](https://github.com/StarRocks/starrocks/pull/32637/commits/5d699f87e0fd163a713a43ef600d3bd829e646b1)